app/scripts/components/analysis/define/index.tsx

```diff
@@ -101,21 +120,170 @@ export default function Analysis() {
   const thematic = useThematicArea();
   if (!thematic) throw resourceNotFound();
 
-  const { date, datasetsLayers, aoi, errors } = useAnalysisParams();
+  const { params, setAnalysisParam } = useAnalysisParams();
+  const { start, end, datasetsLayers, aoi, errors } = params;
 
   const { aoi: aoiDrawState, onAoiEvent } = useAoiControls();
 
+  const [stacSearchStatus, setStacSearchStatus] =
+    useState<ActionStatus>(S_IDLE);
+
   // If there are errors in the url parameters it means that this should be
   // treated as a new analysis. If the parameters are all there and correct, the
   // user is refining the analysis.
   const isNewAnalysis = !!errors?.length;
 
   const analysisParamsQs = analysisParams2QueryString({
-    date,
+    start,
+    end,
     datasetsLayers,
     aoi
   });
 
+  const onStartDateChange = useCallback(
+    (e) => {
+      setAnalysisParam('start', inputFormatToDate(e.target.value));
+    },
+    [setAnalysisParam]
+  );
+
+  const onEndDateChange = useCallback(
+    (e) => {
+      setAnalysisParam('end', inputFormatToDate(e.target.value));
+    },
+    [setAnalysisParam]
+  );
+
+  const allAvailableDatasetsLayers: DatasetLayer[] = useMemo(
+    () =>
+      uniqBy(
+        thematic.data.datasets.map((dataset) => dataset.layers).flat(),
+        'stacCol'
+      ),
+    [thematic.data.datasets]
+  );
+  const selectedDatasetLayerIds = datasetsLayers?.map((layer) => layer.id);
+  const [selectableDatasetLayers, setSelectableDatasetLayers] = useState<
+    DatasetLayer[]
+  >([]);
+  const readyToLoadDatasets = start && end && aoi;
+
+  const onDatasetLayerChange = useCallback(
+    (e) => {
+      const id = e.target.id;
+      const newDatasetsLayers = [...(datasetsLayers || [])];
+      if (e.target.checked) {
+        const newDatasetLayer = allAvailableDatasetsLayers.find(
+          (l) => l.id === id
+        );
+        if (newDatasetLayer) {
+          newDatasetsLayers.push(newDatasetLayer);
+        }
+      } else {
+        const removeAt = newDatasetsLayers.findIndex((l) => l.id === id);
+        newDatasetsLayers.splice(removeAt, 1);
+      }
+      setAnalysisParam('datasetsLayers', newDatasetsLayers);
```

Comment:
It would be better not to use methods that modify the array directly.
`concat` to add, `filter` to remove

Reply:
Since `newDatasetsLayers` is a shallow copy of `datasetsLayers` there's no risk mutating it. But you're right, it's good practice anyways, will fix.